### PR TITLE
Fix issue when calling function with error return in service

### PR DESCRIPTION
**Pull request by riyafa**
## Purpose
Resolve https://github.com/ballerina-platform/ballerina-lang/issues/24795

## Approach
The issue is in generating the method call of a function in a service. Earlier since only resources were allowed in a service the method calls of all functions in a service were throwing errors when it has an error return. Now a check has been added to throw an error only for a resource.

## Samples
N/A

## Remarks
N/A

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/nativeimpl/jvm/tests/MockListener.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                                     latch.countDown();
+                                 }
+
+                                 @Override
+                                 public void notifyFailure(BError error) {
+                                     err = error;
+                                     latch.countDown();
+                                 }
+                             }, new HashMap<>(), null);
+            latch.await();
```
Comment (Kishanthan) [inline tests/jballerina-unit-test/src/test/java/org/ballerinalang/nativeimpl/jvm/tests/MockListener.java:59]: Can you use the BRuntime api's to do this invocation? BExecutor will be removed.

Comment (riyafa) [reply tests/jballerina-unit-test/src/test/java/org/ballerinalang/nativeimpl/jvm/tests/MockListener.java:59]: The BRuntime apis allow executing functions. Can they also be used for executing resources? I found the http code uses the above function

Comment (Kishanthan) [reply tests/jballerina-unit-test/src/test/java/org/ballerinalang/nativeimpl/jvm/tests/MockListener.java:59]: Yes can be done. Thats what @manuranga is changing and http/stdlib will not be using this BExecutor api's anymore and they will be migrated to use BRuntime api's. 

Check the `invokeMethodAsync` method in BRuntime. It should do the above.

Comment (Kishanthan) [reply tests/jballerina-unit-test/src/test/java/org/ballerinalang/nativeimpl/jvm/tests/MockListener.java:59]: And this class BExecutor is no longer there as it has been just removed :)

Comment (riyafa) [reply tests/jballerina-unit-test/src/test/java/org/ballerinalang/nativeimpl/jvm/tests/MockListener.java]: I have fixed this

**Comment by Kishanthan** [comment]
@riyafa There is a build failure in one of the build task. Please check.

**Comment by riyafa** [comment]
That integration test has failed in several PRs. Not sure why.

**Review by Kishanthan — DISMISSED** [review]

**Review by Kishanthan — APPROVED** [review]

**Pull request merged**